docs/src/main/tut/chisel3/memories.md

```diff
@@ -37,41 +37,50 @@ We can create an *n* value sine lookup table using a ROM initialized as follows:
 
 where `amp` is used to scale the fixpoint values stored in the ROM.
 
-## Mem
+## Memories
 
 Memories are given special treatment in Chisel since hardware implementations of memory vary greatly. For example, FPGA memories are instantiated quite differently from ASIC memories. Chisel defines a memory abstraction that can map to either simple Verilog behavioural descriptions or to instances of memory modules that are available from external memory generators provided by foundry or IP vendors.
 
-Chisel supports random-access memories via the `Mem` construct. Writes to `Mem`s are **combinational/asynchronous-read, sequential/synchronous-write**. These `Mem`s will likely be synthesized to register banks, since most SRAMs in modern technologies (FPGA, ASIC) tend to no longer support combinational (asynchronous) reads.
 
-Chisel also has a construct called `SyncReadMem` for **sequential/synchronous-read, sequential/synchronous-write** memories. These `SyncReadMem`s will likely be synthesized to technology SRAMs (as opposed to register banks).
+### `SyncReadMem`: sequential/synchronous-read, sequential/synchronous-write
 
-Ports into Mems are created by applying a `UInt` index.  A 1024-entry register file with one write port and one sequential/synchronous read port might be expressed as follows:
+Chisel has a construct called `SyncReadMem` for sequential/synchronous-read, sequential/synchronous-write memories. These `SyncReadMem`s will likely be synthesized to technology SRAMs (as opposed to register banks).
 
-```scala
-val width:Int = 32
-val addr = Wire(UInt(width.W))
-val dataIn = Wire(UInt(width.W))
-val dataOut = Wire(UInt(width.W))
-val enable = Wire(Bool())
+If the same memory address is both written and sequentially read on the same clock edge, or if a sequential read enable is cleared, then the read data is undefined.
 
-// assign data...
+Values on the read data port are not guaranteed to be held until the next read cycle. If that is the desired behavior, external logic to hold the last read value must be written.
 
-// Create a synchronous-read, synchronous-write memory (like in FPGAs).
-val mem = SyncReadMem(1024, UInt(width.W))
-// Create one write port and one read port.
-mem.write(addr, dataIn)
-dataOut := mem.read(addr, enable)
+#### Read port/write port
+Ports into `SyncReadMem`s are created by applying a `UInt` index.  A 1024-entry register file with one write port and one read port might be expressed as follows:
```

Comment:
```suggestion
Ports into `SyncReadMem`s are created by applying a `UInt` index.  A 1024-entry SRAM with one write port and one read port might be expressed as follows:
```
Or do you mean SRAM/register file? I'm really not sure since the contrast is with `Mem` which we. say usually would be a register file

Reply:
yes sorry, it should be SRAM here.